From 3417e0b57183c64b18a0694089794638a8144e38 Mon Sep 17 00:00:00 2001
From: Yuvaraj Ranganathan <quic_yrangana@quicinc.com>
Date: Wed, 25 Sep 2024 23:11:19 +0530
Subject: [PATCH] PENDING: qcom: ice: Fix incorrect error handling

qcom_ice_create returns error pointer when scm is unavailable but
the return value is checked for NULL pointer.

Signed-off-by: Yuvaraj Ranganathan <quic_yrangana@quicinc.com>
---
 drivers/soc/qcom/ice.c | 8 +-------
 1 file changed, 1 insertion(+), 7 deletions(-)

diff --git a/drivers/soc/qcom/ice.c b/drivers/soc/qcom/ice.c
index 216c6971fa41..3fe4443009bc 100644
--- a/drivers/soc/qcom/ice.c
+++ b/drivers/soc/qcom/ice.c
@@ -66,8 +66,6 @@
 #define qcom_ice_readl(engine, reg)	\
 	readl((engine)->base + (reg))
 
-static bool qcom_ice_create_error;
-
 struct qcom_ice {
 	struct device *dev;
 	void __iomem *base;
@@ -618,14 +616,10 @@ struct qcom_ice *of_qcom_ice_get(struct device *dev)
 	}
 
 	ice = qcom_ice_create(&pdev->dev, base);
-	if (!ice) {
+	if (IS_ERR(ice)) {
 		dev_err(dev, "Cannot get ice instance from %s\n",
 			dev_name(&pdev->dev));
 		platform_device_put(pdev);
-		if (qcom_ice_create_error)
-			ice = ERR_PTR(-EOPNOTSUPP);
-		else
-			ice = ERR_PTR(-EPROBE_DEFER);
 		goto out;
 	}
 
-- 
2.25.1

